webodf/tests/gui/ZoomHelperTests.js

```diff
@@ -0,0 +1,226 @@
+/**
+ * Copyright (C) 2014 KO GmbH <copyright@kogmbh.com>
+ *
+ * @licstart
+ * This file is part of WebODF.
+ *
+ * WebODF is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License (GNU AGPL)
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * WebODF is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with WebODF.  If not, see <http://www.gnu.org/licenses/>.
+ * @licend
+ *
+ * @source: http://www.webodf.org/
+ * @source: https://github.com/kogmbh/WebODF/
+ */
+
+/*global runtime, core, gui, odf, ops, xmldom*/
+
+/**
+ * @constructor
+ * @param {core.UnitTestRunner} runner
+ * @implements {core.UnitTest}
+ */
+gui.ZoomHelperTests = function ZoomHelperTests(runner) {
+    "use strict";
+    var r = runner,
+        t;
+
+    this.setUp = function () {
+        var testArea = core.UnitTest.provideTestAreaDiv();
+        t = {
+            zoomHelper : new gui.ZoomHelper(),
+            scrollPane : testArea,
+            document : testArea.ownerDocument
+        };
+        t.scrollPane.style.position = "relative";
+    };
+
+    this.tearDown = function () {
+        t = {};
+        core.UnitTest.cleanupTestAreaDiv();
+    };
+
+    function setupScrollPane(width, height) {
+        t.scrollPane.style.width = width + "px";
```

Comment:
I'd make these stateless factory methods (e.g., `createScrollPane`) rather than assigning to the test state like this.